File: src/main/java/ru/mm/surv/capture/service/impl/LivenessChecker.java
package ru.mm.surv.capture.service.impl;

import java.util.function.Consumer;

class LivenessChecker implements Runnable {

    private final Process process;
    private final Consumer<Process> action;

    public LivenessChecker(Process process, Consumer<Process> action) {
        this.process = process;
        this.action = action;
    }

    @Override
    public void run() {
        if (!process.isAlive()) {
            action.accept(process);
        }
    }
}

File: backend/src/main/kotlin/ru/mm/surv/capture/ffmpeg/executor/LivenessChecker.kt
package ru.mm.surv.capture.ffmpeg.executor

import java.lang.Process
import java.lang.Runnable
import java.util.function.Consumer

class LivenessChecker(
    private val process: Process,
    private val action: Consumer<Process>) : Runnable {

    override fun run() {
        if (!process.isAlive) {
            action.accept(process)
        }
    }
}
